pkg/reconciler/ingress/controller.go

```diff
@@ -345,3 +354,42 @@ func getSecretInformer(ctx context.Context) v1.SecretInformer {
 	untyped := ctx.Value(filteredFactory.LabelKey{}) // This should always be not nil and have exactly one selector
 	return secretfilteredinformer.Get(ctx, untyped.([]string)[0])
 }
+
+func ensureCtxWithConfigOrDie(ctx context.Context) context.Context {
+	var err error
+	var cfg *store.Config
+	if pollErr := wait.PollUntilContextTimeout(ctx, time.Second, 60*time.Second, true, func(context.Context) (bool, error) {
```

Comment:
Do we want to make the timeout configurable just in case someone has a slow env?

Reply:
Hm I don't think thats necessary. It is only until all yamls are applied when installing (from kubectl or operator). I don't think that should ever be longer than 60 seconds, and if so, it would just restart the pod and you have another 60s.

Reply:
I don't know about what value is good enough tbh, 1m might be enough. 
Restarting is an option but probably you don't want people to have to do that in case they have a big slow cluster (api server under pressure), just in case.
I mean that if they know they have a slow cluster they can configure it high enough (eg. via env var) and forget about it, otherwise they will have to monitor it and restart it.